### PR TITLE
Remove unnecessary intermediate variables

### PR DESCRIPTION
**Pull request by Rubonnek**
In #84445 we've identified a few intermediate variables that are unnecessary. This PR cleans up the code accordingly.

### CODE REVIEW

**Comment by YuriSizov** [comment]
I agree with @Riteo. Cases where it's just used once can be simplified, but others are probably fine (didn't read everything, but things like `received_event` clearly improve readability and aren't useless).

**Comment by YuriSizov** [comment]
Renaming would be fine too. Also cases where it just drops `p_` are fine to remove as well.

File: modules/jpg/image_loader_jpegd.cpp [hunk 1]
```diff
@@ -156,20 +156,19 @@ class ImageLoaderJPGOSBuffer : public jpge::output_stream {
 
 static Error _jpgd_save_to_output_stream(jpge::output_stream *p_output_stream, const Ref<Image> &p_img, float p_quality) {
 	ERR_FAIL_COND_V(p_img.is_null() || p_img->is_empty(), ERR_INVALID_PARAMETER);
-	Ref<Image> image = p_img;
```
[the remark this answers is not in the document]
Comment (Rubonnek) [reply modules/jpg/image_loader_jpegd.cpp:159]: Looked further into this and there's indeed a few bugs here. `p_img` comes from casting a `this` pointer in one of the frames above and the source `Image` data does get mutated:

https://github.com/godotengine/godot/blob/4c96e9676b66d0cc9a25022b019b78f4c20ddc60/core/io/image.cpp#L2529

The return value of the JPG encoder a few lines below is ignored as well, meaning `Image::save_jgp` is oblivious to encoder errors.

Thank you for your comments and feedback in this and my previous PRs. I appreciate it. It has been very valuable.

**Comment by Rubonnek** [comment]
I agree with @KoBeWi and given our differing opinions and the short time I'll be able to dedicate, I don't think I'll be able to properly provide changes we all agree with. For those reasons I'm closing this PR.

**Pull request closed without merging**